onnxruntime/core/providers/qnn/builder/opbuilder/reduce_op_builder.cc

```diff
@@ -188,13 +188,6 @@ Status ReduceOpBuilder::IsOpSupported(QnnModelWrapper& qnn_model_wrapper,
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN EP: ReduceProd operator not supported by HTP backend.");
   }
 
-  if (is_npu_backend) {
-    std::vector<uint32_t> input_shape;
-    ORT_RETURN_IF_NOT(qnn_model_wrapper.GetOnnxShape(node_unit.Inputs()[0].node_arg, input_shape),
-                      "QNN EP: Cannot get input shape for");
-    ORT_RETURN_IF(input_shape.size() > 4, "QNN EP: HTP backend does not support Reduce ops with rank > 4.");
```

Comment:
>input_shape.size() > 4 [](start = 18, length = 22)

What will happen if rank > 5? Will the Op validation fail as expected? Or we still need to keep code here to prevent that happen?

Reply:
It fails as expected on 2.15.1. I added another unit test for this case.